### PR TITLE
[cli] Disable git status background index refresh/optimization

### PR DESCRIPTION
**Pull request by cb1kenobi**
The `create-git-meta` file contains a `isDirty()` function which calls `git status` and subsequently causes one of the test fixture's git index to be updated. Turns out that `git status` will update/optimize the index in the background when invoked: https://git-scm.com/docs/git-status#_background_refresh.

By disabling this using the `--no-optional-locks` flag, `isDirty()` will no longer update the index when checking the status.

From my testing, the `--no-optional-locks` flag must be set after `git` and before `status`.

This PR is likely a better alternative to https://github.com/vercel/vercel/pull/8568 as this PR eliminates the problem all together.

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [ ] This PR has a concise title and thorough description useful to a reviewer
- [ ] Issue from task tracker has a link to this PR

### CODE REVIEW

File: packages/cli/src/util/create-git-meta.ts [hunk 1]
```diff
@@ -78,16 +78,20 @@ function getLastCommit(directory: string): Promise<git.Commit> {
...
   return new Promise(resolve => {
-    exec('git status -s', { cwd: directory }, function (err, stdout, stderr) {
-      let debugMessage = `Failed to determine if Git repo has been modified:`;
-      if (err || stderr) {
-        if (err) debugMessage += `\n${err}`;
-        if (stderr) debugMessage += `\n${stderr.trim()}`;
-        output.debug(debugMessage);
-        return resolve(false);
+    exec(
+      'git --no-optional-locks status -s',
```
Comment (EndangeredMassa) [inline packages/cli/src/util/create-git-meta.ts:85]: **suggestion (non-blocking):** add comment about why `--no-optional-locks` is necessary

Can we add a comment about this flag? The flag name and its [official description](https://git-scm.com/docs/git/2.22.1#Documentation/git.txt---no-optional-locks) don't do a great job explaining what the actual effects are.

**Review by EndangeredMassa — APPROVED** [review]

File: packages/cli/src/util/create-git-meta.ts (hunk 1 above)
Comment (cb1kenobi) [reply packages/cli/src/util/create-git-meta.ts:85]: Good idea! Done!

**Review by EndangeredMassa — APPROVED** [review]

**Review by Ethan-Arrowood — APPROVED** [review]
lgtm!

File: packages/cli/src/util/create-git-meta.ts (hunk 1 above)
Comment (styfle) [reply packages/cli/src/util/create-git-meta.ts:85]: If we're worried about the order of the flags, we could use the `GIT_OPTIONAL_LOCKS=0` env var as an alternative

**Review by styfle — APPROVED** [review]

**Comment by cb1kenobi** [comment]
@styfle We have another option... we could set `GIT_OPTIONAL_LOCKS=1` just for the unit test. That way we don't need to set the `--no-optional-locks` flag in production and we don't have to worry about the test mutating files. Thoughts?

**Comment by styfle** [comment]
@cb1kenobi I think its worth keeping for improved performance for our users too. We just want the quickest way to know if the working tree is dirty or not.

**Review by TooTallNate — APPROVED** [review]

**Pull request merged**